mm2src/mm2_main/src/lp_ordermatch/best_orders.rs

```diff
@@ -223,6 +225,8 @@ pub async fn best_orders_rpc(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>,
     let mut response = HashMap::new();
     if let Some((p2p_response, peer_id)) = best_orders_res {
         log::debug!("Got best orders {:?} from peer {}", p2p_response, peer_id);
+        let my_pubsecp = mm2_internal_pubkey_hex(&ctx).map_err(|mm2_err| mm2_err.to_string())?;
+        let orderbook = ordermatch_ctx.orderbook.lock();
```

Comment:
Instead of locking the orderbook in every loop, wouldn't it be better to either:

1 - clone `orderbook.my_p2p_pubkeys` before entering the loop by using
```rust
let my_p2p_pubkeys = ordermatch_ctx.orderbook.lock().my_p2p_pubkeys.clone();
```
2 - using my maker orders to check for the order uuid instead, also locking and cloning before entering the loop
```rust
let my_maker_orders_uuids = ordermatch_ctx
    .maker_orders_ctx
    .lock()
    .orders
    .keys()
    .cloned()
    .collect::<HashSet<_>>();
```
I vote for the second option since `orderbook` is used a lot more than `maker_orders_ctx` which means locking it less is better.

Reply:
I see the idea is in to not keep a mutex locked during the all request processing by copying the set of order ids [2] or pubkeys if that is `ZHTLC` [1].

The second option is more universal and straightforward for any chain and it look attractive )

If you let me I would keep a possibility  to determine `is_mine` [just a single comparison](https://github.com/KomodoPlatform/atomicDEX-API/blob/1015-best-orders-is_mine/mm2src/mm2_main/src/lp_ordermatch.rs#L367)  but not retrieving it from hash set in most cases.

P.S.: Just to mention. If it is needed further we could use bloom filter to determine if order is mine or not
P.S.S.: In terms of locking context there are quite lot to improve to not keep it locked during request processing in big variety of ways.

Reply:
> If you let me I would keep a possibility to determine is_mine [just a single comparison](https://github.com/KomodoPlatform/atomicDEX-API/blob/1015-best-orders-is_mine/mm2src/mm2_main/src/lp_ordermatch.rs#L367) but not retrieving it from hash set in most cases.

Sure, it looks good either way, `my_zhtlc_orders_pubkeys` should be a smaller set than maker orders in most cases which makes the comparison faster.

> P.S.: Just to mention. If it is needed further we could use bloom filter to determine if order is mine or not

Thanks for the suggestion, I don't think it's needed now since a specific number of best orders can be requested meaning that the returned set will not be large if needed. We can consider it in the future if we have a need for it.

> P.S.S.: In terms of locking context there are quite lot to improve to not keep it locked during request processing in big variety of ways.

Can you please elaborate more on this?

Reply:
Like as you suggested the first way is look over the all places where ctx is locked and copy data or user lock free methods, that is what I thought, but it's presumably over optimizing )

Reply:
> Like as you suggested the first way is look over the all places where ctx is locked and copy data

We already did that in a PR in the past, now we have to only be careful when adding new code that they don't cause deadlocks or lock a mutex when the lock is not needed, sometimes we keep the lock because the functionality of the code requires the data to not change while working on it.